heat/sparse/factories.py

```diff
@@ -0,0 +1,254 @@
+"""Provides high-level Dcsr_matrix initialization functions"""
+
+import torch
+import numpy as np
+from scipy.sparse import csr_matrix as scipy_csr_matrix
+
+from typing import Optional, Type, Union
+import warnings
+
+from ..core.communication import MPI, sanitize_comm, Communication
+from ..core.devices import Device
+from ..core.types import datatype
+
+from .dcsr_matrix import Dcsr_matrix
+
+from ..core import devices
+from ..core import types
+
+__all__ = [
+    "sparse_csr_matrix",
+]
+
+
+def sparse_csr_matrix(
```

Comment:
Mimics `ht.array` function but for sparse csr matrices. Accepts a `torch.sparse_csr_tensor` or `scipy.sparse.csr_matrix` and builds the distributed csr matrix (`Dcsr_matrix`).